core/sql/exp/exp_datetime.cpp

```diff
@@ -2337,6 +2430,50 @@ convertStrToMonth(char * value, char *result)
   return FALSE;
 }
 
+static short convSrcDataToDst(Lng32 numSrcBytes, char* &srcData, 
+                               Lng32 numTgtBytes, char *dstData,
+                               const char * nextByte,
+                               CollHeap * heap, ComDiagsArea** diagsArea)
+{
+  Lng32 src = 0;
+  Lng32 val = 0;
+  for (val = 0, src = 0; src < numSrcBytes && isDigit8859_1(*srcData); 
+       src++, srcData++)
+    val = val * 10 + (*srcData - '0');
+  
+  if (src < numSrcBytes) 
+    {
+      // string contains non-digit
+      //
+      ExRaiseSqlError(heap, diagsArea, EXE_CONVERT_DATETIME_ERROR);
+      return -1;
+    }
+
+  if (numTgtBytes == sizeof(Lng32))
+    *(Lng32*)dstData = val;
+  else if (numTgtBytes == sizeof(short))
+    *(short*)dstData = val;
+  else if (numTgtBytes == sizeof(char))
+    *(char*)dstData = val;
+  else 
```

Comment:
Never mind. Looks like we depend on the caller to make sure conditions are such that overflows will not occur.